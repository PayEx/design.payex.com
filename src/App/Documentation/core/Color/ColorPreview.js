import React from "react";
import PropTypes from "prop-types";

const ColorPreview = ({ name, hex }) => {
    const style = { backgroundColor: hex };

    return (
        <div className="color-preview">
            <div className={`color-block${name === "white" ? " border" : ""}`} style={style}/>
            <p>
                {name}
                <small><br/>HEX: {hex}</small>
            </p>
        </div>
    );
};

ColorPreview.propTypes = {
    name: PropTypes.string.isRequired,
    hex: PropTypes.string.isRequired
};

export default ColorPreview;
