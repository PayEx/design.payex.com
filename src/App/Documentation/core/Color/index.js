import React from "react";

import { ComponentPreview, DocContainer } from "#";
import ColorPreview from "./ColorPreview";

const MainColors = () => (
    <>
        <h2 id="main-colors">Main colors</h2>
        <p>The main color palette of Swedbank Pay</p>
        <ComponentPreview language="html" showCasePanel>
            <div className="row">
                <div className="col-xs-12 col-sm-6 col-md-4 col-lg-2 offset-lg-1 mb-2">
                    <ColorPreview name="orange-medium" hex="#ee7023" />
                </div>
                <div className="col-xs-12 col-sm-6 col-md-4 col-lg-2 mb-2">
                    <ColorPreview name="orange-light" hex="#ff9900" />
                </div>
                <div className="col-xs-12 col-sm-6 col-md-4 col-lg-2 mb-2">
                    <ColorPreview name="orange-bright" hex="#ffcc00" />
                </div>
                <div className="col-xs-12 col-sm-6 col-md-4 col-lg-2 mb-2">
                    <ColorPreview name="green" hex="#82b236" />
                </div>
                <div className="col-xs-12 col-sm-6 col-md-4 col-lg-2 mb-2">
                    <ColorPreview name="blue" hex="#5b8ad6" />
                </div>
                <div className="col-xs-12 col-sm-6 col-md-4 col-lg-2 offset-lg-1 mb-2">
                    <ColorPreview name="blue-light" hex="#a2c9ce" />
                </div>
                <div className="col-xs-12 col-sm-6 col-md-4 col-lg-2 mb-2">
                    <ColorPreview name="blue-bright" hex="#d7e6e9" />
                </div>
                <div className="col-xs-12 col-sm-6 col-md-4 col-lg-2 mb-2">
                    <ColorPreview name="teal" hex="#4893a0" />
                </div>
                <div className="col-xs-12 col-sm-6 col-md-4 col-lg-2 mb-2">
                    <ColorPreview name="blue-bright-alt" hex="#a3c9cf" />
                </div>
                <div className="col-xs-12 col-sm-6 col-md-4 col-lg-2 mb-2">
                    <ColorPreview name="red-medium" hex="#dd3300" />
                </div>
            </div>
        </ComponentPreview>
    </>
);

const AlertColors = () => (
    <>
        <h2 id="alert-colors">Alert colors</h2>
        <ComponentPreview language="html" showCasePanel>
            <div className="row">
                <div className="col-12">
                    <ColorPreview name="brand-success" hex="#477d17" />
                </div>
                <div className="col-12">
                    <ColorPreview name="brand-success-light" hex="#f2f7eb" />
                </div>
                <div className="col-12">
                    <ColorPreview name="brand-warning" hex="#ff9900" />
                </div>
                <div className="col-12">
                    <ColorPreview name="brand-warning-light" hex="#fffae5" />
                </div>
                <div className="col-12">
                    <ColorPreview name="brand-error-border" hex="#f1bbb9" />
                </div>
                <div className="col-12">
                    <ColorPreview name="brand-error" hex="#b24843" />
                </div>
                <div className="col-12">
                    <ColorPreview name="brand-error-light" hex="#fceeed" />
                </div>
                <div className="col-12">
                    <ColorPreview name="brand-info" hex="#2c7a8f" />
                </div>
                <div className="col-12">
                    <ColorPreview name="brand-info-light" hex="#f0f9fb" />
                </div>
            </div>
        </ComponentPreview>
    </>
);

const Grayscale = () => (
    <>
        <h2 id="grayscale">Grayscale</h2>
        <ComponentPreview language="html" showCasePanel>
            <div className="row">
                <div className="col-12">
                    <ColorPreview name="black" hex="#000000" color="#eaeaea" />
                </div>
                <div className="col-12">
                    <ColorPreview name="nearblack" hex="#111111" color="#eaeaea" />
                </div>
                <div className="col-12">
                    <ColorPreview name="gray-dark" hex="#262626" color="#eaeaea" />
                </div>
                <div className="col-12">
                    <ColorPreview name="gray-medium-dark" hex="#363636" color="#eaeaea" />
                </div>
                <div className="col-12">
                    <ColorPreview name="gray-medium" hex="#404040" color="#eaeaea" />
                </div>
                <div className="col-12">
                    <ColorPreview name="gray-medium-light" hex="#545454" color="#eaeaea" />
                </div>
                <div className="col-12">
                    <ColorPreview name="gray-light" hex="#666666" color="#eaeaea" />
                </div>
                <div className="col-12">
                    <ColorPreview name="gray-light-alt" hex="#999999" />
                </div>
                <div className="col-12">
                    <ColorPreview name="gray-bright" hex="#eaeaea" />
                </div>
                <div className="col-12">
                    <ColorPreview name="gray-bright-alt" hex="#bcbcbc" />
                </div>
                <div className="col-12">
                    <ColorPreview name="neargray" hex="#f9f9f9" />
                </div>
                <div className="col-12">
                    <ColorPreview name="offwhite" hex="#f4f4f4" />
                </div>
                <div className="col-12">
                    <ColorPreview name="nearwhite" hex="#f0f0f0" />
                </div>
                <div className="col-12">
                    <ColorPreview name="white" hex="#ffffff" color="black" />
                </div>
            </div>
        </ComponentPreview>
    </>
);

const Color = () => (
    <DocContainer>
        <p className="lead">The Swedbank Pay DesignGuide color palette</p>
        <MainColors />
        <AlertColors />
        <Grayscale />
    </DocContainer>
);

export default Color;

/* For testing */
export { MainColors, AlertColors, Grayscale };
