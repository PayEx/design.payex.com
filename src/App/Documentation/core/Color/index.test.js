import React from "react";
import { shallow } from "enzyme";

import Color, { MainColors, AlertColors, Grayscale } from "./index";

describe("Core: Color", () => {
    it("is defined", () => {
        expect(Color).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Color />);

        expect(wrapper).toMatchSnapshot();
    });

    it("MainColors renders", () => {
        const wrapper = shallow(<MainColors />);

        expect(wrapper).toMatchSnapshot();
    });

    it("AlertColors renders", () => {
        const wrapper = shallow(<AlertColors />);

        expect(wrapper).toMatchSnapshot();
    });

    it("Grayscale renders", () => {
        const wrapper = shallow(<Grayscale />);

        expect(wrapper).toMatchSnapshot();
    });
});
