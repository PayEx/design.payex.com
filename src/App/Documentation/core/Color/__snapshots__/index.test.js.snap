// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Core: Color AlertColors renders 1`] = `
<Fragment>
  <h2
    id="alert-colors"
  >
    Alert colors
  </h2>
  <ComponentPreview
    language="html"
    showCasePanel={true}
  >
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <ColorPreview
          hex="#477d17"
          name="brand-success"
        />
      </div>
      <div
        className="col-12"
      >
        <ColorPreview
          hex="#f2f7eb"
          name="brand-success-light"
        />
      </div>
      <div
        className="col-12"
      >
        <ColorPreview
          hex="#ff9900"
          name="brand-warning"
        />
      </div>
      <div
        className="col-12"
      >
        <ColorPreview
          hex="#fffae5"
          name="brand-warning-light"
        />
      </div>
      <div
        className="col-12"
      >
        <ColorPreview
          hex="#f1bbb9"
          name="brand-error-border"
        />
      </div>
      <div
        className="col-12"
      >
        <ColorPreview
          hex="#b24843"
          name="brand-error"
        />
      </div>
      <div
        className="col-12"
      >
        <ColorPreview
          hex="#fceeed"
          name="brand-error-light"
        />
      </div>
      <div
        className="col-12"
      >
        <ColorPreview
          hex="#2c7a8f"
          name="brand-info"
        />
      </div>
      <div
        className="col-12"
      >
        <ColorPreview
          hex="#f0f9fb"
          name="brand-info-light"
        />
      </div>
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Core: Color Grayscale renders 1`] = `
<Fragment>
  <h2
    id="grayscale"
  >
    Grayscale
  </h2>
  <ComponentPreview
    language="html"
    showCasePanel={true}
  >
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <ColorPreview
          color="#eaeaea"
          hex="#000000"
          name="black"
        />
      </div>
      <div
        className="col-12"
      >
        <ColorPreview
          color="#eaeaea"
          hex="#111111"
          name="nearblack"
        />
      </div>
      <div
        className="col-12"
      >
        <ColorPreview
          color="#eaeaea"
          hex="#262626"
          name="gray-dark"
        />
      </div>
      <div
        className="col-12"
      >
        <ColorPreview
          color="#eaeaea"
          hex="#363636"
          name="gray-medium-dark"
        />
      </div>
      <div
        className="col-12"
      >
        <ColorPreview
          color="#eaeaea"
          hex="#404040"
          name="gray-medium"
        />
      </div>
      <div
        className="col-12"
      >
        <ColorPreview
          color="#eaeaea"
          hex="#545454"
          name="gray-medium-light"
        />
      </div>
      <div
        className="col-12"
      >
        <ColorPreview
          color="#eaeaea"
          hex="#666666"
          name="gray-light"
        />
      </div>
      <div
        className="col-12"
      >
        <ColorPreview
          hex="#999999"
          name="gray-light-alt"
        />
      </div>
      <div
        className="col-12"
      >
        <ColorPreview
          hex="#eaeaea"
          name="gray-bright"
        />
      </div>
      <div
        className="col-12"
      >
        <ColorPreview
          hex="#bcbcbc"
          name="gray-bright-alt"
        />
      </div>
      <div
        className="col-12"
      >
        <ColorPreview
          hex="#f9f9f9"
          name="neargray"
        />
      </div>
      <div
        className="col-12"
      >
        <ColorPreview
          hex="#f4f4f4"
          name="offwhite"
        />
      </div>
      <div
        className="col-12"
      >
        <ColorPreview
          hex="#f0f0f0"
          name="nearwhite"
        />
      </div>
      <div
        className="col-12"
      >
        <ColorPreview
          color="black"
          hex="#ffffff"
          name="white"
        />
      </div>
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Core: Color MainColors renders 1`] = `
<Fragment>
  <h2
    id="main-colors"
  >
    Main colors
  </h2>
  <p>
    The main color palette of Swedbank Pay
  </p>
  <ComponentPreview
    language="html"
    showCasePanel={true}
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12 col-sm-6 col-md-4 col-lg-2 offset-lg-1 mb-2"
      >
        <ColorPreview
          hex="#ee7023"
          name="orange-medium"
        />
      </div>
      <div
        className="col-xs-12 col-sm-6 col-md-4 col-lg-2 mb-2"
      >
        <ColorPreview
          hex="#ff9900"
          name="orange-light"
        />
      </div>
      <div
        className="col-xs-12 col-sm-6 col-md-4 col-lg-2 mb-2"
      >
        <ColorPreview
          hex="#ffcc00"
          name="orange-bright"
        />
      </div>
      <div
        className="col-xs-12 col-sm-6 col-md-4 col-lg-2 mb-2"
      >
        <ColorPreview
          hex="#82b236"
          name="green"
        />
      </div>
      <div
        className="col-xs-12 col-sm-6 col-md-4 col-lg-2 mb-2"
      >
        <ColorPreview
          hex="#5b8ad6"
          name="blue"
        />
      </div>
      <div
        className="col-xs-12 col-sm-6 col-md-4 col-lg-2 offset-lg-1 mb-2"
      >
        <ColorPreview
          hex="#a2c9ce"
          name="blue-light"
        />
      </div>
      <div
        className="col-xs-12 col-sm-6 col-md-4 col-lg-2 mb-2"
      >
        <ColorPreview
          hex="#d7e6e9"
          name="blue-bright"
        />
      </div>
      <div
        className="col-xs-12 col-sm-6 col-md-4 col-lg-2 mb-2"
      >
        <ColorPreview
          hex="#4893a0"
          name="teal"
        />
      </div>
      <div
        className="col-xs-12 col-sm-6 col-md-4 col-lg-2 mb-2"
      >
        <ColorPreview
          hex="#a3c9cf"
          name="blue-bright-alt"
        />
      </div>
      <div
        className="col-xs-12 col-sm-6 col-md-4 col-lg-2 mb-2"
      >
        <ColorPreview
          hex="#dd3300"
          name="red-medium"
        />
      </div>
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Core: Color renders 1`] = `
<DocContainer>
  <p
    className="lead"
  >
    The Swedbank Pay DesignGuide color palette
  </p>
  <MainColors />
  <AlertColors />
  <Grayscale />
</DocContainer>
`;
